### ISSUE TITLE
Implementing the given architecture

### ISSUE THREAD

**Issue by RohitSaha** [keras-team/keras#9539]
<img width="418" alt="screen shot 2018-03-02 at 1 44 58 pm" src="https://user-images.githubusercontent.com/13007024/36918582-5d2ae94a-1e28-11e8-8766-d732f9126e64.png">

The idea is to have a shared model that accepts labeled and unlabeled inputs. If inputs are **labeled**, we calculate the crossentropy loss. An extra domain loss is calculated which is essentially the mmd between features of the labeled input and unlabeled input computed by the adapt layer (as mentioned in the image). This domain loss is then added to the crossentropy loss. 

I looked into Keras's Sequential API guide and have managed to implement some portion of the code. I am not sure how to proceed with getting outputs from specific layers of the model based on the input tensor and then calculating the domain loss. Code I've implemented so far:

    model = load_model(weight_path)

	inp = model.input
	out = model.layers[-2].output
	adapt_layer = Dense(512, activation='sigmoid', name='adapt_layer')(out) //adapt layer
	preds = Dense(19, activation='softmax', name='predictions', activity_regularizer=regularizers.l2(0.01))(adapt_layer)
	vision_model = Model(inputs=inp, outputs=preds)

	labeled_input = Input(shape=(None, 1024, 1024, 3))
	unlabeled_input= Input(shape=(None, 1024, 1024, 3))

	preds_labeled = vision_model(labeled_input) //output for calculating crossentropy loss

	labeled_features = vision_model(labeled_input).layers[-2].output //not sure if this is the way to go
	unlabeled_features = vision_model(unlabeled_input).layers[-2].output

Any help would be appreciated!

**Comment by RohitSaha** [comment]
I got some clarification and was able to implement the following code:
 
    model = load_model(weight_path)
   
    //The part of the architecture common to both labeled and unlabeled data
    inp = model.input
    out = model.layers[-2].output 
    adapt_layer = Dense(512, activation='sigmoid', name='adapt_layer')(out) //adapt layer
    vision_model = Model(inputs=inp, outputs=preds)

    //Different inputs
    labeled_input = Input(shape=(None, 1024, 1024, 3)) 
    unlabeled_input= Input(shape=(None, 1024, 1024, 3))

    //getting outputs of the adapt layer based on the input data
    labeled_adapt_features = vision_model(labeled_input)
    unlabeled_adapt_features = vision_model(unlabeled_input)

    //Predictions will be made only for the labeled dataset, hence only labeled_features is considered.
    predictions = Dense(19, activation='softmax', name='predictions', activity_regularizer=regularizers.l2(0.01))(labeled_features)
   
    classification_model = Model([labeled_input, unlabeled_input], predictions)


Can anyone say if the code is correct for the architecture in the diagram (posted above)? Thanks in advance!

**Comment by nio747** [comment]
Hello, I wonder if you have already solved this issue? I am trying to implement this method in Keras and get stucked here too. Have you thought about customizing a MMD regularizer?

**Comment by tobilab** [comment]
> Hello, I wonder if you have already solved this issue? I am trying to implement this method in Keras and get stucked here too. Have you thought about customizing a MMD regularizer?

Dear @nio747 and @RohitSaha  i am currently trying the same as you last year, i try to implement the network above. I could net get it done with the code from @RohitSaha  above so i wanted to ask you if one of you tackled the implementation of the network above in keras and would share the changed running code to reproduce it?

this would help very much. thank you in advice, tobias
